File: ui/src/protocol/Mode.js
import PropTypes from 'prop-types';

const Mode = {
  NONE:{
    code: 0,
  },
  DELETE:{
    code: 1,
  },
  CREATE:{
    code: 2,
  },
  MODIFY:{
    code: 3,
  },

}
export default Mode;

export const valueOf = (code) => Object.values(Mode).find(r => r.code === code);

valueOf.propTypes = {
  code: PropTypes.number.isRequired
}

File: log/src/main/java/com/lookforlog/protocol/constants/Mode.java
package com.lookforlog.protocol.constants;

import java.util.Arrays;

public enum Mode {
    NONE(0x0),
    DELETE(0x1),
    CREATE(0x2),
    MODIFY(0x3),;

    /**
     * size of mode, byte
     */
    public static final int SIZE = 1;
    private int flag;

    Mode(int flag) {
        this.flag = flag;
    }

    public static Mode valueOf(int flag) {
        return Arrays.stream(Mode.values()).filter(m -> m.flag == flag).findFirst().orElse(null);
    }

    public int getFlag() {
        return flag;
    }
}
